project/src/Agents/GSTNode2DAgent.gd

```diff
@@ -0,0 +1,165 @@
+		angular_velocity = body.angular_velocity
+
+
+func _apply_sliding_steering(accel: Vector3) -> void:
+	var velocity := GSTUtils.to_vector2(linear_velocity + accel).clamped(linear_speed_max)
+	if apply_linear_drag:
+		velocity = velocity.linear_interpolate(Vector2.ZERO, linear_drag_percentage)
+	velocity = body.move_and_slide(velocity)
+	if calculate_velocities:
+		linear_velocity = GSTUtils.to_vector3(velocity)
+
+
+func _apply_collide_steering(accel: Vector3, delta: float) -> void:
+	var velocity := GSTUtils.clampedv3(linear_velocity + accel, linear_speed_max)
+	if apply_linear_drag:
+		velocity = velocity.linear_interpolate(
+				Vector3.ZERO,
+				linear_drag_percentage
+		)
+	body.move_and_collide(GSTUtils.to_vector2(velocity) * delta)
+	if calculate_velocities:
+		linear_velocity = velocity
+
+
+func _apply_normal_steering(accel: Vector3, delta: float) -> void:
+	var velocity := GSTUtils.clampedv3(linear_velocity + accel, linear_speed_max)
+	if apply_linear_drag:
+		velocity = velocity.linear_interpolate(
+				Vector3.ZERO,
+				linear_drag_percentage
+		)
+	body.global_position += GSTUtils.to_vector2(velocity) * delta
+	if calculate_velocities:
+		linear_velocity = velocity
+
+
+func _apply_orientation_steering(angular_acceleration: float, delta: float) -> void:
+	var velocity = angular_velocity + angular_acceleration
+	if apply_angular_drag:
+		velocity = lerp(velocity, 0, angular_drag_percentage)
+	body.rotation += velocity * delta
+	if calculate_velocities:
+		angular_velocity = velocity
+
+
+func _set_use_physics(value: bool) -> void:
+	if use_physics and not value:
+		body.get_tree().disconnect("idle_frame", self, "_on_SceneTree_frame")
+		body.get_tree().connect("physics_frame", self, "_on_SceneTree_frame")
+	elif not use_physics and value:
+		body.get_tree().disconnect("physics_frame", self, "_on_SceneTree_frame")
+		body.get_tree().connect("idle_frame", self, "_on_SceneTree_frame")
+	use_physics = value
+
+
+func _set_body(value: Node2D) -> void:
+	body = value
+	if body is RigidBody2D:
+		_body_type = BodyType.RIGID
+	elif body is KinematicBody2D:
+		_body_type = BodyType.KINEMATIC
+	else:
+		_body_type = BodyType.NODE
+	
+	if _body_type == BodyType.RIGID:
+		linear_velocity = GSTUtils.to_vector3(body.linear_velocity)
+		angular_velocity = body.angular_velocity
+	else:
+		_last_position = body.global_position
+		_last_orientation = body.rotation
+	
+	position = GSTUtils.to_vector3(_last_position)
+	orientation = _last_orientation
+
+
+func _on_body_ready() -> void:
+	body.get_tree().connect("physics_frame", self, "_on_SceneTree_frame")
+	_set_body(body)
+
+
+func _on_SceneTree_frame() -> void:
+	var current_position: Vector2 = body.global_position
+	var current_orientation: float = body.rotation
+	
+	position = GSTUtils.to_vector3(current_position)
+	orientation = current_orientation
+	
+	if calculate_velocities:
+		if _applied_steering:
+			_applied_steering = false
+		else:
+			match _body_type:
+				BodyType.RIGID:
```

Comment:
That's typically the kind of code that makes me wonder if it wouldn't be better to have separate classes for KinematicBody, RigidBody, making naming clearer, and also so that we could add them as needed. Right now, I'm not sure it's a good thing to allow the user to work with any kind of node, and to have to support that in the framework, especially if none of us is using or needs it.

In short, it's the You Ain't Gonna Need it principle. I'm not sure it applies right now, but I'd like your thoughts on that. Are you already using rigidbodies in the demos or somewhere with the framework?